test/index.spec.ts

```diff
@@ -1,54 +1,65 @@
-import Vue from 'vue'
-import Vuex, { mapState } from 'vuex'
-import VueRouter from 'vue-router'
-import { sync } from '@/index'
+import { createApp, defineComponent, h, computed, nextTick } from 'vue'
 
-Vue.use(Vuex)
-Vue.use(VueRouter)
+import { createStore, useStore } from 'vuex'
+import { createRouter, createWebHistory, RouterView } from 'vue-router'
+import { sync } from '@/index'
 
-function run(originalModuleName: string, done: Function): void {
+async function run(originalModuleName: string, done: Function): Promise<void> {
   const moduleName: string = originalModuleName || 'route'
 
-  const store = new Vuex.Store({
-    state: { msg: 'foo' }
+  const store = createStore({
+    state() {
+      return { msg: 'foo' }
+    }
   })
 
-  const Home = Vue.extend({
-    computed: mapState(moduleName, {
-      path: (state: any) => state.fullPath,
-      foo: (state: any) => state.params.foo,
-      bar: (state: any) => state.params.bar
-    }),
-    render(h) {
-      return h('div', [this.path, ' ', this.foo, ' ', this.bar])
+  const Home = defineComponent({
+    setup() {
+      const store = useStore()
+      const path = computed(() => store.state[moduleName].fullPath)
+      const foo = computed(() => store.state[moduleName].params.foo)
+      const bar = computed(() => store.state[moduleName].params.bar)
+      return () => h('div', [path.value, ' ', foo.value, ' ', bar.value])
     }
   })
 
-  const router = new VueRouter({
-    mode: 'abstract',
-    routes: [{ path: '/:foo/:bar', component: Home }]
+  const router = createRouter({
+    history: createWebHistory(),
+    routes: [
+      {
+        path: '/',
+        component: {
+          template: 'root'
+        }
+      },
+      { path: '/:foo/:bar', component: Home }
+    ]
   })
 
   sync(store, router, {
     moduleName: originalModuleName
   })
 
   router.push('/a/b')
+  await router.isReady()
   expect((store.state as any)[moduleName].fullPath).toBe('/a/b')
   expect((store.state as any)[moduleName].params).toEqual({
     foo: 'a',
     bar: 'b'
   })
 
-  const app = new Vue({
-    store,
-    router,
-    render: (h) => h('router-view')
-  }).$mount()
+  const rootEl = document.createElement('div')
```

Comment:
Wouldn't it be easier and safer to use vue test utils?

Reply:
Ok, I will try to use vue test utils!
(Actually, I have not use it yet 😅 )